### PR TITLE
Lberg/semantic pose

### PR DESCRIPTION
**Pull request by lucabergamini**
Add the following:
- refactor `SemanticRasterizer` and protobuf utility function to work with world coordinates and not local ENU;
- use `world_to_image` matrix into `SemanticRasterizer`
- reduce ROI for lane search to half the size of the hypotenuse from the image centre (2x performance)
- add some tests

### CODE REVIEW

File: l5kit/l5kit/data/map.py [hunk 1]
```diff
@@ -54,37 +54,32 @@ def proto_to_semantic_map(map_fragment: "MapFragment") -> dict:
...
+            xyz_left = unpack_boundary(lane.left_boundary, lane.geo_frame, ecef_to_pose)
+            xyz_right = unpack_boundary(lane.right_boundary, lane.geo_frame, ecef_to_pose)
+            lanes.append({"xyz_left": xyz_left, "xyz_right": xyz_right, "id": element.id.id.decode("utf-8")})
+
             # store bounds for fast rasterisation look-up
-            x_min = min(np.min(x_left), np.min(x_right))
-            y_min = min(np.min(y_left), np.min(y_right))
-            x_max = max(np.max(x_left), np.max(x_right))
-            y_max = max(np.max(y_left), np.max(y_right))
+            x_min = min(np.min(xyz_left[:, 0]), np.min(xyz_right[:, 0]))
```
Comment (yye-lyft) [inline l5kit/l5kit/data/map.py:62]: What's benefits of packing (x,y,z) together? I prefer the old way which separete them, it is easier to understand

File: l5kit/l5kit/rasterization/semantic_rasterizer.py [hunk 2]
```diff
@@ -2,49 +2,53 @@
...
-    rx = (np.cos(yaw) * (px - x) - np.sin(yaw) * (py - y)) / pixel_size[0] + 0.5 * raster_size[0]
-    ry = (np.sin(yaw) * (px - x) + np.cos(yaw) * (py - y)) / pixel_size[1] + 0.5 * raster_size[1]
-    return rx.astype(int) * 256, ry.astype(int) * 256
 
-
-def elements_within_radius(x: float, y: float, bounds: np.ndarray, radius: float) -> np.ndarray:
+def elements_within_bounds(center: np.ndarray, bounds: np.ndarray, half_side: float) -> np.ndarray:
     """
-    Get indices of elements for which bounds are inside a radius from center (x,y)
+    Get indices of elements for which the bounding box described by bounds intersect the one defined around
```
Comment (oliver-scheel) [inline l5kit/l5kit/rasterization/semantic_rasterizer.py:17]: "intersectS", and I would put ("square with side length ...")

File: l5kit/l5kit/rasterization/semantic_rasterizer.py [hunk 3]
```diff
@@ -92,66 +98,57 @@ def rasterize(
             self.raster_size, self.pixel_size, ego_translation, ego_yaw, self.ego_center,
         )
 
-        # get pose of center pixel
+        # get XY of center pixel in world coordinate
```
Comment (oliver-scheel) [inline l5kit/l5kit/rasterization/semantic_rasterizer.py:101]: coordinates

File: l5kit/l5kit/tests/rasterization/sem_rasterizer_test.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import numpy as np
+
+from l5kit.rasterization.semantic_rasterizer import elements_within_bounds
+
+
+def test_elements_within_bounds() -> None:
+    center = np.zeros(2)
+    half_side = 0.5  # square centered on origin with side 1
```
Comment (oliver-scheel) [inline l5kit/l5kit/tests/rasterization/sem_rasterizer_test.py:8]: around

File: l5kit/l5kit/rasterization/semantic_rasterizer.py [hunk 5]
```diff
@@ -2,49 +2,53 @@
...
     Returns:
         np.ndarray: indices of elements inside radius from center
     """
-    x_min_in = x > bounds[:, 0, 0] - radius
-    y_min_in = y > bounds[:, 0, 1] - radius
-    x_max_in = x < bounds[:, 1, 0] + radius
-    y_max_in = y < bounds[:, 1, 1] + radius
+    x_center, y_center = center
+
+    x_min_in = x_center > bounds[:, 0, 0] - half_side
```
Comment (yye-lyft) [inline l5kit/l5kit/rasterization/semantic_rasterizer.py:30]: `half_extent`

**Pull request merged**